bundles/CoreBundle/EventListener/SearchBackendListener.php

```diff
@@ -87,4 +87,21 @@ public function onPostUpdateElement(ElementEventInterface $e)
             new SearchBackendMessage(Service::getElementType($element), $element->getId())
         );
     }
+
+
+    /**
+     * @param ElementEventInterface $e
+     */
+    public function onPostUpdateDataObjectDocument(ElementEventInterface $e)
+    {
+        //do not update index when auto save or only saving version
+        if (
+            ($e->hasArgument('isAutoSave') && $e->getArgument('isAutoSave')) ||
+            ($e->hasArgument('saveVersionOnly') && $e->getArgument('saveVersionOnly'))
+        ) {
+            return;
+        }
+
+        $this->onPostAddElement($e);
```

Comment:
Why does `onPostUpdateDataObjectDocument` call `onPostAddElement` and not `onPostUpdateElement`?

Reply:
Hmm, maybe because `onPostAddElement` and `onPostUpdateElement` are the same...

What about having only one method for `onPostAddElement`, `onPostUpdateElement` and `onPostUpdateDataObjectDocument` as all three are basically the same?

Reply:
Changed it `onPostAddUpdateElement` for all elements.